plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -0,0 +1,332 @@
+                COLUMN_PROJECTION_RANGE,
+                this::splitCommaSeparatedString);
+        rewriteProperty(
+                metastoreTableProperties,
+                trinoTableProperties,
+                getMetastoreProjectionPropertyKey(columnName, PROPERTY_SUFFIX_COLUMN_PROJECTION_INTERVAL),
+                COLUMN_PROJECTION_INTERVAL,
+                Integer::valueOf);
+        rewriteProperty(
+                metastoreTableProperties,
+                trinoTableProperties,
+                getMetastoreProjectionPropertyKey(columnName, METASTORE_PROPERTY_KEY_SUFFIX_COLUMN_PROJECTION_INTERVAL_UNIT),
+                COLUMN_PROJECTION_INTERVAL_UNIT,
+                value -> ChronoUnit.valueOf(value.toUpperCase(Locale.ENGLISH)));
+        rewriteProperty(
+                metastoreTableProperties,
+                trinoTableProperties,
+                getMetastoreProjectionPropertyKey(columnName, PROPERTY_SUFFIX_COLUMN_PROJECTION_DIGITS),
+                COLUMN_PROJECTION_DIGITS,
+                Integer::valueOf);
+        rewriteProperty(
+                metastoreTableProperties,
+                trinoTableProperties,
+                getMetastoreProjectionPropertyKey(columnName, PROPERTY_SUFFIX_COLUMN_PROJECTION_FORMAT),
+                COLUMN_PROJECTION_FORMAT,
+                value -> value);
+        return trinoTableProperties;
+    }
+
+    private Projection parseColumnProjection(String columnName, Type columnType, Map<String, Object> columnProperties)
+    {
+        ProjectionType projectionType = (ProjectionType) columnProperties.get(COLUMN_PROJECTION_TYPE);
+        if (Objects.isNull(projectionType)) {
+            throw projectionException("Projection type property missing for column: '" + columnName + "'");
+        }
+        ProjectionFactory projectionFactory = Optional.ofNullable(projectionFactories.get(projectionType))
+                .orElseThrow(() -> projectionException(format("Projection type %s for column: '%s' not supported", projectionType, columnName)));
+        if (!projectionFactory.isSupportedColumnType(columnType)) {
+            throw unsupportedProjectionColumnTypeException(columnName, columnType);
+        }
+        return projectionFactory.create(columnName, columnType, columnProperties);
+    }
+
+    private <I, V> void rewriteProperty(
+            Map<String, I> sourceProperties,
+            Map<String, V> targetProperties,
+            String sourcePropertyKey,
+            String targetPropertyKey,
+            Function<I, V> valueMapper)
+    {
+        Optional.ofNullable(sourceProperties.get(sourcePropertyKey))
+                .ifPresent(value -> targetProperties.put(targetPropertyKey, valueMapper.apply(value)));
+    }
+
+    private TrinoException projectionException(String message)
+    {
+        return new TrinoException(StandardErrorCode.INVALID_COLUMN_PROPERTY, format("Partition projection failure. %s", message));
```

Comment:
I do no think having `Partition projection failure. ` prefix makes those exceptions more readable. I would just craft the sentence for each place where you use `projectionException` to state clearly that the context is "partitions projections", yet in a shape which matches nicely to specific error being reported.

Reply:
Ok applied